### PR TITLE
Introduce `Market` struct

### PR DESCRIPTION
**Pull request by nlordell**
This PR introduces a new `Market` structure for specifying base and quote tokens so there is no ambiguity in the order in which they are specified.

### Test Plan

No significant changes, unit tests should catch any logic errors that may have been introduced (that may accidentally reverse the order of the ask and bid orders).

### CODE REVIEW

File: pricegraph/src/lib.rs [hunk 1]
```diff
@@ -84,6 +84,37 @@ impl TransitiveOrder {
     }
 }
 
+/// A struct representing a market.
+///
+/// This is used for computing transitive orderbooks.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
+pub struct Market {
+    /// The base token being purchased in a market.
```
Comment (fleupold) [inline pricegraph/src/lib.rs:93]: Technically the base token can also be sold in a market, against the bids.

File: pricegraph/src/lib.rs [hunk 2]
```diff
@@ -84,6 +84,37 @@ impl TransitiveOrder {
...
+    pub base: TokenId,
+    /// The quote token being sold in a market. Prices in a market are always
+    /// expressed in the quote token.
+    pub quote: TokenId,
+}
+
+impl Market {
+    /// Returns the token pair for ask orders.
+    pub fn ask_pair(self) -> TokenPair {
```
Comment (fleupold) [inline pricegraph/src/lib.rs:102]: It's a bit counter-intuitive to me that you can turn a TokenPair into one market but one market into two token pairs. Maybe we should clarify the `Into` more explicitly into something like from_buy_order?

**Review by fleupold — APPROVED** [review]
Small note about <Into> being unique but <From> not, otherwise looks good.

File: pricegraph/src/lib.rs (hunk 1 above)
Comment (nlordell) [reply pricegraph/src/lib.rs:93]: I really wasn't sure how to describe base and quote tokens... I'll look up a Wikipedia definition or something.

File: pricegraph/src/lib.rs (hunk 2 above)
Comment (nlordell) [reply pricegraph/src/lib.rs:102]: Oh! You mean the `price_estimator::models::TokenPair`. I'll rename that, I think that type should be a "market" anyway.

**Review by e00E — APPROVED** [review]

File: pricegraph/src/lib.rs (hunk 1 above)
Comment (nlordell) [reply pricegraph/src/lib.rs:93]: I'm going to use language from: https://en.wikipedia.org/wiki/Currency_pair

File: pricegraph/src/lib.rs (hunk 2 above)
Comment (nlordell) [reply pricegraph/src/lib.rs:102]: @fleupold  Hey, I refactored the `price_estimator::models::TokenPair` to not use `Into<pricegraph::TokenPair>` and `Into<pricegraph::Market>` which was admittedly a bit confusing. Its now `price_estimator::models::Market` and implements `Deref<Target = Market>` and we have a `Filter` prefix that gets the bid token pair for a market. Let me know if this is a little clearer.

**Pull request merged**